docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,540 @@
+:code:`ocn_tend_thick` is called. This is stored in 
+:code:`layerThickness(tlev=2)`.
+
+There is a new remapping routine with inputs:
+
+- :code:`layerThickness(tlev=2)` for scalars
+- :code:`layerThicknessEdge(tlev=2)` for velocity
+- :code:`bottomDepth`
+- :code:`layerThicknessTarget`
+- :code:`layerThicknessEdgeTarget`
+- Remapped and updated: :code:`normalVelocity`
+- Remapped and updated: All members of :code:`tracerPool` unless 
+  :code:`activeTracersOnly`, in which case only the :code:`activeTracers`
+- *This may not be a complete list*
+
+This routine makes calls to the PPR library
+
+*More details here*
+
+Some implementation considerations for PPR:
+ 
+- Error-checking in PPR: make consistent with MPAS errors, consider additional
+  error checks
+- *Add more here*
+
+:code:`layerThickness(tlev=2)` is then overwritten with the new layer thickness.
+
+After remapping, :code:`ocn_diagnostic_solve` is called. This is needed to 
+compute the density and pressure fields based on the remapped ocean state and
+the diagnostic field :code:`vertVelocityTop` which is the vertical velocity 
+through the top of the layer. This is only used as a diagnostic variable for 
+computing the MOC streamfunction. None of the mixing parameterizations require
+a vertical velocity (Eulerian or diasurface velocity).
+
+Note: if `vertVelocityTop` is computed between regridding and remapping then it 
+can be computed as 
+
+.. code::
+   
+   vertVelocityTop(k) = vertVelocityTop(k+1) - div_hu(k) - 
+                        (layerThickness(k,tlev=2) - layerThickness(k,tlev=1))/dt
+
+If `vertVelocityTop` is computed after remapping, then :code:`div_hu` is no
+longer appropriate as it has been remapped. In this case, the Lagrangian layer 
+thickness should be stored in a scratch variable and then the vertical velocity 
+through the top of the layer can be computed:
+
+.. code::
+
+   layerThicknessALE = layerThickness(tlev=2)
+   
+   layerThickness(tlev=2) = layerThicknessTarget
+   
+   vertVelocityTop = (layerThickness(tlev=2) - layerThicknessALE)/dt
+
+The computation of :code:`vertTransportVelocityTop` and 
+:code:`vertGMBolusVelocitytop` is unchanged as these fields represent Eulerian 
+velocities.
+
+Implementation: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+After determining the target grid, perform the following steps:
+
+#. Apply damping function to limit the rate of grid movement in one timestep 
+   [and enforce a maximum movement limit?]. 
+#. Optional: Assign :math:`h_k^{t+1}` to :math:`h_k^{lg}` if 
+   :math:`h_k^{t+1} - h_k^{lg}` is less than a minimum thickness alteration. 
+   This motivated by accuracy considerations, as each remapping may introduce 
+   errors. *Darren, would this improve PPR computational performance?*
```

Comment:
@dengwirda Would this provide any benefits for PPR's accuracy/efficiency?

Reply:
This is an interesting idea to investigate! As you say, if there's insufficient difference between the Lagrangian grid and the (pending) target grid z^{t+1}, it may be worthwhile to simply skip the remapping step. This should be both cheaper, and (may) reduce the overall amount of numerical dissipation. 

As above, the time/space smoothing for z^{t+1} should only come into play for non-diagnostic target grid choices.

Reply:
> As above, the time/space smoothing for z^{t+1} should only come into play for non-diagnostic target grid choices.

Smoothing discussion removed by https://github.com/MPAS-Dev/MPAS-Model/commit/6c1f1a91d466c47f049d3c4db0cc46ad7f9d585d

Reply:
I could imagine that remapping at irregular intervals could also induce strange behavior like larger shocks to the system each time remapping does occur.

I wonder if this might be something to put on the back burner and explore for future improvements in performance and/or accuracy.